tests/integration/runPrettier.js

```diff
@@ -113,11 +113,60 @@ function runPrettier(dir, args = [], options = {}) {
     jest.restoreAllMocks();
   }
 
-  const result = { status, stdout, stderr, write };
+  return { status, stdout, stderr, write };
+
+  function appendStdout(text) {
+    if (status === undefined) {
+      stdout += text;
+    }
+  }
+  function appendStderr(text) {
+    if (status === undefined) {
+      stderr += text;
+    }
+  }
+}
+
+let hasRunningCli = false;
+function runPrettier(dir, args = [], options = {}) {
+  let promise;
+  const getters = {
+    get status() {
+      return runCli().then(({ status }) => status);
+    },
+    get stdout() {
+      return runCli().then(({ stdout }) => stdout);
+    },
+    get stderr() {
+      return runCli().then(({ stderr }) => stderr);
+    },
+    get write() {
+      return runCli().then(({ write }) => write);
+    },
+    test: testResult,
+  };
 
-  const testResult = (testOptions) => {
-    for (const name of Object.keys(result)) {
-      test(`(${name})`, () => {
+  return getters;
+
+  function runCli() {
+    if (hasRunningCli) {
+      throw new Error("Please wait for previous CLI to exit.");
+    }
+
+    if (!promise) {
+      hasRunningCli = true;
+      promise = run(dir, args, options);
+      promise.then(() => {
+        hasRunningCli = false;
+      });
```

Comment:
```suggestion
      promise = run(dir, args, options).finally(() => {
        hasRunningCli = false;
      });
```
This way it's simpler to understand when exactly `hasRunningCli = false;` is executed.